### PR TITLE
Add ability for clients to forward local connections

### PR DESCRIPTION
**Pull request by ibuildthecloud**
## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [ ] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: cmd/client.go [hunk 1]
```diff
@@ -68,12 +69,17 @@ func runClient(cmd *cobra.Command, _ []string) error {
 	log.Printf("%s", WelcomeMessage)
 	log.Printf("Starting client - version %s", getVersion())
 
+	listeners, err := cmd.Flags().GetStringArray("listen")
```
Comment (alexellis) [inline cmd/client.go:72]: Should `listen` be `local` here to match the new flag on line 23?

**Comment by alexellis** [comment]
Hi @ibuildthecloud did you use an upstream tag or release of remotedialer for this?

@jsiebens are you interested in testing it out?

**Comment by jsiebens** [comment]
Hi @alexellis 

Sure, I just gave it a try, and it is working perfectly.
A specific use case that you want me to try?

**Comment by alexellis** [comment]
I think we need to have a release tag upstream @ibuildthecloud in order to be able to use the feature. It sounds like the implementation is working as expected 👍

Thank you for testing Johan, can you include the command like parameters you used and what you tested?

**Comment by jsiebens** [comment]
The test was done with two VM in the same VPC

On the first VM, install docker and run the echoserver 
`docker run -t -i --rm -p 8090:8080 k8s.gcr.io/echoserver:1.4`

On the second VM, install this version of inlets and start an inlets server:
 `inlets server --token=<token>`

On my local machine, run the inlets client and forward port 3000:
 `inlets client --remote=128.199.56.91:8000 --local=3000:10.133.0.15:8080 --token=<token>`
In this case, 128.199.56.91 is the public IP of the first VM and 10.133.0.15 is the private IP of the second VM.

And test with curl:

```
$ curl -i http://localhost:3000
HTTP/1.1 200 OK
Content-Type: text/plain
Date: Sat, 10 Oct 2020 10:33:10 GMT
Server: nginx/1.10.0
Content-Length: 334

CLIENT VALUES:
client_address=10.133.0.14
command=GET
real path=/
query=nil
request_version=1.1
request_uri=http://10.133.0.15:8080/

SERVER VALUES:
server_version=nginx: 1.10.0 - lua: 10001

HEADERS RECEIVED:
accept=*/*
accept-encoding=gzip
host=10.133.0.15
user-agent=curl/7.68.0
x-forwarded-for=127.0.0.1
BODY:
```

File: cmd/client.go (hunk 1 above)
Comment (alexellis) [reply cmd/client.go:72]: @ibuildthecloud this is a minor error, probably after renaming, but I needed to change it when testing.

**Comment by alexellis** [comment]
Thank you both for the input.. I'm going to apply the patches and release this. I think we may also want a slightly stricter ACL option for the server, to say which services can be forwarded back to the client from the server's network.

**Comment by alexellis** [comment]
In addition `--local=3000:10.133.0.15:8080` binds to 127.0.0.1:3000 on the client, but some users may want an option for `0.0.0.0:3000` on the client, so I'm going to take a look on hacking on that too.

Thanks again @ibuildthecloud

**Pull request closed without merging**